lib/puppet/module/task.rb

```diff
@@ -157,7 +193,23 @@ def implementations
     end
 
     def files
-      implementations.map {|imp| { 'name' => imp['name'], 'path' => imp['path'] } }
+      md = metadata
+      lib_files = []
+      impl_lib_files = []
+
+      unless md.nil?
+        lib_files = md.key?('files') ? self.class.find_files(md['files'], @module) : []
+        # There's definitely a more elegant way to do this...
+        if md.key?('implementations')
+          impl_files = []
+          md['implementations'].map { |impl| impl_files << impl['files'] if impl.key?('files') }
```

Comment:
This could be `each` instead of `map`. Also I'd do the `uniq` on the combination of `impl_lib_files` and `lib_files`.

Reply:
So maybe add `impl_files` to `lib_files`, then `uniq!`, then `self.class.find_files`.

Reply:
This still feels messy - namely having to instantiate all of these as empty arrays at the top. I'm a sucker for one-liners though

Reply:
We can certainly make it a one-liner.